sourcegraph/about
# Performance Improvements and bug fixes

This PR provides overall performance improvements and increases our Lighthouse score.

### Changelog
- updated next.js
- added Plausible analytics, which closes #5801 (1.3 KB)
- updated CustomerLogos component to match DLS with new props and assets, which closes #5807
  - the component now tailors to various container widths
  - dark mode and monochrome mode are now available
  - updated mercado libre logo (old logo was 15 KB, new logo is 5 KB)
  - updated databricks logo (size is the same ~4 KB)
- added a Heading component for valid ordering of headings when needing design-specific sizes but need to render as another tag for valid, semantic, a11y accessibly ordering. Closes #5808
- updated valid order of headings in IntegrationsSection and footer
- added passive to scroll event listener for performance
- added `next-pwa` for a minimalist PWA config which generates our service worker. While this isn't super useful yet, it increases our Lighthouse score. 8-)
- added production source maps via recommendation for best practices and production troubleshooting
- added manifest, logo marks, and document meta for iOS and PWA requirements
- added explicit image sizes and alt tag updates for banner, footer, and homepage hero per lighthouse recommendations
- YouTube CORS error fixes which was degrading performance from console errors
- updated common breakpoints to use Tailwind breakpoints from theme
- compressed mesh background images for homepage hero which was degrading performance for our LCP (32% decrease in file size)
- added Drift facade for lazy loading Drift integration, which before, was negatively impacting performance

### Test

1. Ensure prettier has standardized the proposed changes.
2. Ensure Plausible is recording analytics events
3. Ensure Drift chat loads after clicking the facade. Ensure it lazy loads from cache after refresh.
4. Ensure saving the site as a PWA (add to homescreen) via Safari in iOS works
5. Ensure CustomerLogos section resizes accordingly (look at smaller container variant on contact pages)
6. Ensure valid sequentially descending ordering of heading elements on homepage (not all pages are using Heading yet)

### Metrics

This is using Chrome Dev Tools Lighthouse Report (Desktop device)

![Screen Shot 2022-09-30 at 1 34 29 AM](https://user-images.githubusercontent.com/1733936/193197569-2336b3ca-46bb-4a28-b000-2b810ff06929.png)
